lib/utils/updateQueue.hpp

```diff
@@ -160,9 +160,10 @@ struct fmt::formatter<updateQueue<T>> {
 
     template<typename FormatContext>
     auto format(const updateQueue<T>& q, FormatContext& ctx) {
-        auto it = q.get_all_updates().begin();
+        auto all_updates = q.get_all_updates();
+        auto it = all_updates.begin();
         auto pos = ctx.out();
-        while (it != q.get_all_updates().end()) {
+        while (it != all_updates.end()) {
             pos = format_to(pos, "{:f} ", ts_to_double(it->first));
             *it++;
```

Comment:
Looks good, and the correct fix to me.

My only comment is that it's probably neater using a range based for loop. It seems I can't apply a suggestion, but something like:

```C++
auto pos = ctx.out();

for (const auto& item : q.get_all_updates()) {
    pos = format_to(pos, "{:f} ", ts_to_double(item.first));
}
```

That's mostly a preference thing, so ignore if you want.